bigframes/dtypes.py

```diff
@@ -214,6 +214,16 @@ def is_orderable(type: ExpressionType) -> bool:
     return not is_array_like(type) and not is_struct_like(type) and (type != GEO_DTYPE)
 
 
+def is_clusterable(type: ExpressionType) -> bool:
+    # https://cloud.google.com/bigquery/docs/clustered-tables#cluster_column_types
+    # This is based on default database type mapping, could in theory represent in non-default bq type to cluster.
+    return (
+        not is_array_like(type)
+        and not is_struct_like(type)
+        and (type not in (GEO_DTYPE, TIME_DTYPE, FLOAT_DTYPE))
```

Comment:
Geo is clusterable but not orderable.

Reply:
Should we make this an allowlist, instead? I suspect as new types are added they aren't likely to be clusterable.

Reply:
added clusterable property to dtype metadata struct, defaulting as `False`